pageserver/src/bin/pageserver.rs

```diff
@@ -459,11 +460,18 @@ fn start_pageserver(conf: &'static PageServerConf) -> Result<()> {
         }
     }
 
-    // Initialize tenant manager.
-    tenant_mgr::init(conf);
-
     // keep join handles for spawned threads
-    let mut join_handles = vec![];
+    // don't spawn threads before daemonizing
+    let mut join_handles = Vec::new();
+
+    let storage_accessor = relish_storage::init_storage(conf)?.map(|(accessor, handle)| {
+        join_handles.push(handle);
+        accessor
+    });
+    // Initialize tenant manager.
+    if let Some(handle) = tenant_mgr::init(conf, storage_accessor) {
```

Comment:
This is a supplementary thread to listen to the events from the main upload/download thread.
This is needed and needed in the `tenant_mgr` since apparently this is the only place where the actual repositories are stored: I need to add images into those repos or create new repos, if absent.

This thread lives only as long as the listener channel counterpart inside `storage_accessor` does. That one stops doing anything and exits the moment its sender channel counterpart is dropped.
That one is contained in the main upload/download thread, so presumably, after we deal with the graceful main upload/download thread shutdown, we're automatically shutting this thread down.